src/accordion-pane/index.tsx

```diff
@@ -63,24 +68,46 @@ export const AccordionPane = factory(function LoadingIndicator({
 		};
 	};
 
-	const initialOpen = (key: string) => {
+	const open = (key: string | number) => {
 		const openKeys: AccordionPaneICache['openKeys'] = icache.get('openKeys') || {};
-		return openKeys[key];
+		return !!openKeys[key];
 	};
 
+	const Pane = icache.getOrSet('Pane', () =>
+		paneFactory(function Pane({
+			children: paneChildren,
+			id,
+			properties: paneProperties,
+			middleware: { theme: paneTheme }
+		}) {
+			const { key = id } = paneProperties();
+
+			const localTheme = paneTheme.compose(
+				titlePaneCss,
+				css,
+				'pane'
+			);
+
+			console.log(localTheme);
+
+			return (
+				<div classes={paneTheme.variant()}>
+					<TitlePane
+						key={key}
+						onClose={onClose(key)}
+						onOpen={onOpen(key)}
+						open={open(key)}
+						theme={localTheme}
+					>
+						{paneChildren()[0]}
+					</TitlePane>
+				</div>
+			);
+		})
+	);
```

Comment:
@tomdye: The `AccordionPane` dynamically creates this `Pane` to return to its child renderer. This has to be dynamically created since it uses `onClose`, `onOpen`, and `open` functions that are local to `AccordionPane`. All works fine, except passing the composed theme to the child `TitlePane` on line 100. Everything looks correct, but `TitlePane`, for whatever reason, doesn't pick up and use the classes. Oddly, if I manually delete the `variant` property on that `localTheme` composed theme object, everything works.

Any thoughts on what I need to do differently?

Reply:
Paul I simply meant you'd create a new widget inside the accordion pane index.tsx called `Pane` that did the theme.compose onto the titlepane or simply wrapped them in a div or something. It wouldn't be injected into the renderer, the user would just import it from the file like we do with breadcrumb / breadcrumbSeparator. The user would still have to do the open / close checks etc via the callbacks, this was meant to simply remove the `theme` step.